### PR TITLE
Feature/pattern lib

### PR DESCRIPTION
**Pull request by phivk**
first version of very basic pattern lib documentation of components

### CODE REVIEW

File: app/Http/routes.php [hunk 1]
```diff
@@ -237,6 +237,44 @@ function()
...
     });
 
+    Route::get('patternlib', function () {
+        $COMPONENT_BASE_PATH = '../resources/views/components/';
+        
+        // get json file names
+        $file_names = array();
+        $handle=opendir($COMPONENT_BASE_PATH);
+        while (false !== ($file_name = readdir($handle))):
+            if(substr($file_name, -5) == '.json'):
```
Comment (igor-kamil) [inline app/Http/routes.php]: what is "-5" ?

File: app/Http/routes.php [hunk 2]
```diff
@@ -237,6 +237,44 @@ function()
         return view('dielo', array('item' => $item, 'more_items' => $more_items, 'previous' => $previous, 'next' => $next));
     });
 
+    Route::get('patternlib', function () {
+        $COMPONENT_BASE_PATH = '../resources/views/components/';
```
Comment (igor-kamil) [inline app/Http/routes.php]: nice point to follow const convention with uppercase
why not to go even further and define it as const? -> `const COMPONENT_BASE_PATH` within class definitions

File: app/Http/routes.php [hunk 3]
```diff
@@ -237,6 +237,44 @@ function()
         return view('dielo', array('item' => $item, 'more_items' => $more_items, 'previous' => $previous, 'next' => $next));
     });
 
+    Route::get('patternlib', function () {
```
Comment (igor-kamil) [inline app/Http/routes.php]: wouldn't it be nicer to use separated controller for `patternlib` ?

File: app/Http/routes.php (hunk 1 above)
Comment (phivk) [reply app/Http/routes.php]: `substr($file_name, -5)` returns the last 5 character of the string $file_name
if this is equal to ".json" we know we are dealing with a JSON file

Comment (rastislav-chynoransky) [reply app/Http/routes.php]: there is more descriptive helper function in laravel named `ends_with`

File: app/Http/routes.php [hunk 4]
```diff
@@ -237,6 +237,44 @@ function()
...
+        $COMPONENT_BASE_PATH = '../resources/views/components/';
+        
+        // get json file names
+        $file_names = array();
+        $handle=opendir($COMPONENT_BASE_PATH);
+        while (false !== ($file_name = readdir($handle))):
+            if(substr($file_name, -5) == '.json'):
+                $file_names[] = $file_name;
+            endif;
+        endwhile;
```
Comment (rastislav-chynoransky) [inline app/Http/routes.php]: this is just a detail (and we should definitely use linter in future), but please don't mix those two synatxes

File: app/Http/routes.php [hunk 5]
```diff
@@ -237,6 +237,44 @@ function()
...
+        foreach ($file_paths as $file_path) {
+            $file_str = file_get_contents($file_path);
+            $file_json = json_decode($file_str, true);
+            $components_json[] = $file_json;
+        }
+
+        // add attributes
+        foreach ($components_json as $index => $component) {
+            $components_json[$index]['include_path'] = 'components.'.substr($file_names[$index], 0, -5);
+            $components_json[$index]['source_code'] = htmlspecialchars(file_get_contents(substr($file_paths[$index], 0, -5).'.blade.php'));
```
Comment (rastislav-chynoransky) [inline app/Http/routes.php]: is this `htmlspecialchars` necessary? isn't it what blade's {{ }} macro(?) does?

File: app/Http/routes.php (hunk 4 above)
Comment (igor-kamil) [reply app/Http/routes.php]: try to keep following PSR-2 as coding standard - refer to http://www.php-fig.org/psr/psr-2/#11-example

**Comment by igor-kamil** [comment]
please update the changelog following http://keepachangelog.com/en/1.0.0/

File: app/Http/routes.php (hunk 5 above)
Comment (phivk) [reply app/Http/routes.php]: you're right:
https://laravel.com/docs/5.5/blade#displaying-data

**Pull request merged**